packages/in_app_purchase/in_app_purchase_storekit/ios/Classes/FIAObjectTranslator.m

```diff
@@ -167,15 +167,33 @@ + (NSDictionary *)getMapFromNSError:(NSError *)error {
   if (!error) {
     return nil;
   }
+
+  NSString *unsupportedMessage =
+      @"Unable to encode native userInfo object of type %@ to map. Please submit an issue at "
+      @"https://github.com/flutter/flutter/issues/new with the title \"[in_app_purchase_storekit] "
+      @"Unable to encode userInfo of type %@\" and add reproduction steps and the error details in "
+      @"the description field.";
+
   NSMutableDictionary *userInfo = [NSMutableDictionary new];
   for (NSErrorUserInfoKey key in error.userInfo) {
     id value = error.userInfo[key];
     if ([value isKindOfClass:[NSError class]]) {
       userInfo[key] = [FIAObjectTranslator getMapFromNSError:value];
     } else if ([value isKindOfClass:[NSURL class]]) {
       userInfo[key] = [value absoluteString];
+    } else if ([value isKindOfClass:[NSArray class]]) {
+      NSMutableArray *errors = [[NSMutableArray alloc] init];
+      for (id error in value) {
+        if ([error isKindOfClass:[NSError class]]) {
+          [errors addObject:[FIAObjectTranslator getMapFromNSError:error]];
+        } else {
+          [errors addObject:[NSString
+                                stringWithFormat:unsupportedMessage, [value class], [value class]]];
+        }
+      }
+      userInfo[key] = errors;
     } else {
```

Comment:
You'll need to handle strings as well; that's what broke the existing test.

Reply:
Fixed, I also included a check on `NSNumber` just in case (including matching unit test).